src/wolfssl_binding.rs

```diff
@@ -72,39 +91,54 @@ impl<S: Read + Write> SslStream<S> {
             method: ManuallyDrop::new(method),
             _p: PhantomData,
         })
-        */
     }
 }
-/*  [TODO] Copied pasted from openssl_binding:
-pub fn create_openssl_client(
-    stream: MemoryStream,
-    tls_version: &TLSVersion,
-) -> Result<SslStream<MemoryStream>, ErrorStack> {
-    let mut ctx_builder = SslContext::builder(SslMethod::tls())?;
-    // Not sure whether we want this disabled or enabled: https://gitlab.inria.fr/mammann/tlspuffin/-/issues/26
-    // The tests become simpler if disabled to maybe that's what we want. Lets leave it default
-    // for now.
-    // https://wiki.openssl.org/index.php/TLS1.3#Middlebox_Compatibility_Mode
-    #[cfg(feature = "openssl111")]
-    ctx_builder.clear_options(SslOptions::ENABLE_MIDDLEBOX_COMPAT);
-
-    set_max_protocol_version(&mut ctx_builder, tls_version)?;
 
-    // Disallow EXPORT in client
-    ctx_builder.set_cipher_list("ALL:!EXPORT:!LOW:!aNULL:!eNULL:!SSLv2")?;
-
-    let mut ssl = Ssl::new(&ctx_builder.build())?;
-    ssl.set_connect_state();
-
-    SslStream::new(ssl, stream)
+unsafe fn set_max_protocol_version(
+    ctx: *mut wolf::WOLFSSL_CTX,
+    tls_version: &TLSVersion,
+) -> Result<(), ErrorStack> {
+    unsafe {
+        match tls_version {
+            TLSVersion::V1_3 => {
+                #[cfg(feature = "openssl111")]
+                wolf::wolfSSL_CTX_set_max_proto_version(ctx, wolf::TLS1_3_VERSION as i32);
+                // do nothing as the maximum available TLS version is 1.3
+                Ok(())
+            }
+            TLSVersion::V1_2 => {
+                wolf::wolfSSL_CTX_set_max_proto_version(ctx, wolf::TLS1_2_VERSION as i32);
+                Ok(())
+            }
+            TLSVersion::Unknown => Ok(()),
+        }?;
+    }
+    Ok(())
 }
- */
 
 pub fn create_client(
     stream: MemoryStream,
     tls_version: &TLSVersion,
 ) -> Result<SslStream<MemoryStream>, ErrorStack> {
-    todo!()
+    unsafe {
+        //// Global WolfSSL lib initialization
+        init();
+
+        //// Context builder
+        wolf::wolfSSL_Init();
+        let ctx: *mut wolf::WOLFSSL_CTX = wolf::wolfSSL_CTX_new(wolf::wolfTLSv1_3_client_method());
+        // Disallow EXPORT in client
+        let cipher_list = CString::new("ALL:!EXPORT:!LOW:!aNULL:!eNULL:!SSLv2").unwrap();
+        wolf::wolfSSL_CTX_set_cipher_list(ctx, cipher_list.as_ptr() as *const _); //
```

Comment:
Things are simpler than I thought thanks to the OpenSSL compatibility later of WolfSSL that I finally managed to enable (see `build.rs` in the wolfssl-sys crate and the option `--enable-all`).

Reply:
Good :)